services/src/main/java/org/fao/geonet/api/mapservers/GeoServerRest.java

```diff
@@ -323,13 +323,17 @@ public boolean deleteCoverage(String cs, String c) throws IOException {
     /**
      * @param ws Name of the workspace to put the datastore in
      * @param ds Name of the datastore
-     * @param f  Zip {@link java.io.File} to upload containing a shapefile
+     * @param f  Zip {@link java.io.File} to upload containing a shapefile, or a geopackage
      */
-    public boolean createDatastore(String ws, String ds, Path f) throws IOException {
-        Log.debug(Geonet.GEOPUBLISH, "Creating datastore " + ds + " in workspace " + ws + " from path " + f.toString());
+    public boolean createDatastore(String ws, String ds, Path f, String format) throws IOException {
+        String contentType = "application/x-sqlite3";
+        if (f.getFileName().toString().toLowerCase().endsWith(".zip")) {
```

Comment:
`toString()` should not be required, no?

Reply:
i'm totally not a regular java developer, but `getFileName()` returns a `Path` object so i think it's needed ?

Reply:
You are right sorry, I though it was returning a `String`